pkg/report/handler.go

```diff
@@ -58,8 +58,21 @@ func ShowHandler() func(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
+
 		objStat, err := obj.Stat()
-		if nil == err && "application/x-tar" == objStat.ContentType {
+		// not found error can be a directory
+		if objErr, _ := err.(minio.ErrorResponse); http.StatusNotFound == objErr.StatusCode {
```

Comment:
Why we are not returning the error here although it couldn't be found but still serving the file? Serving won't break because the file doesn't exist?

Reply:
>Why we are not returning the error here although it couldn't be found but still serving the file? Serving won't break because the file doesn't exist?

Not really, `http.FileServer(fs)` will look for `index.html` inside the directory (if is actually a directory) and proper handler in case not find anything

>Regarding to your comment, shouldn't we check if request is a directory ?

There is no way to check if is a directory or not, actually, S3 does not have directory concept at all.